pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/ProcessRuntime.java

```diff
@@ -139,15 +139,19 @@ public void start() {
 
         // Note: we create the expected log folder before the function process logger attempts to create it
         // This is because if multiple instances are launched they can encounter a race condition creation of the dir.
-        log.info("Creating function log directory {}", funcLogDir);
-        boolean success = createFolder(funcLogDir);
 
-        if (!success) {
-            log.error("Log folder could not be created : {}", funcLogDir);
-            throw new RuntimeException("Log folder creation error");
-        }
+        final File dir = new File(funcLogDir);
+        if (!dir.exists()) {
+            log.info("Creating function log directory {}", funcLogDir);
+            boolean success = createFolder(funcLogDir);
```

Comment:
Instead of this, isn't there a way to check for the create folder exception to handle atomically the error condition?

Reply:
if indeed that was an exception something else down the line will throw an exception

Reply:
add most I would just have an warning message but not throw the exception